sklearn/manifold/tests/test_t_sne.py

```diff
@@ -954,7 +985,64 @@ def test_tsne_square_distances_futurewarning(metric, square_distances):
         assert not record
 
 
+@pytest.mark.parametrize('init', [None, 'random', 'pca'])
+# This test can be deleted in 1.2
+def test_tsne_init_futurewarning(init):
+    # Make sure that a FutureWarning is only raised when the
+    # init is not specified or is 'pca'
+    random_state = check_random_state(0)
+
+    X = random_state.randn(5, 2)
+    kwargs = dict(learning_rate=200.0, init=init)
+    tsne = TSNE(**{k: v for k, v in kwargs.items() if v is not None})
+
+    if init is None:
+        with pytest.warns(FutureWarning, match="The default initialization.*"):
+            tsne.fit_transform(X)
+    elif init == 'pca':
+        with pytest.warns(FutureWarning, match="The PCA initialization.*"):
+            tsne.fit_transform(X)
+    else:
+        with pytest.warns(None) as record:
+            tsne.fit_transform(X)
+        assert not record
+
+
+@pytest.mark.parametrize('learning_rate', [None, 200.0])
+# This test can be deleted in 1.2
+def test_tsne_learning_rate_futurewarning(learning_rate):
+    # Make sure that a FutureWarning is only raised when the learning rate
+    # is not specified
+    random_state = check_random_state(0)
+
+    X = random_state.randn(5, 2)
+    kwargs = dict(learning_rate=learning_rate, init='random')
+    tsne = TSNE(**{k: v for k, v in kwargs.items() if v is not None})
+
+    if learning_rate is None:
+        with pytest.warns(FutureWarning, match="The default learning rate.*"):
+            tsne.fit_transform(X)
+    else:
+        with pytest.warns(None) as record:
+            tsne.fit_transform(X)
+        assert not record
+
+
+@pytest.mark.parametrize('sample_size', [500, 10000])
```

Comment:
This test is too long, we have to keep the testing suite as short as possible. If a faster test is not possible, I would remove this test.
> [============================= slowest 20 durations =============================](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=26885&view=logs&j=13650d7c-49e8-54f3-0598-c3480d1c1e4f&t=97f2162d-a0c4-54dc-9fce-2fceef7172e3&l=353)
**226.91s call     manifold/tests/test_t_sne.py::test_auto_learning_rate[10000]**
14.74s call     tests/test_common.py::test_estimators[SequentialFeatureSelector(estimator=LogisticRegression(C=1))-check_estimator_sparse_data]
13.22s call     ensemble/tests/test_voting.py::test_gridsearch
10.09s call     svm/tests/test_svm.py::test_svc_ovr_tie_breaking[NuSVC]
9.17s call     utils/tests/test_estimator_checks.py::test_check_estimator_clones
**7.92s call     manifold/tests/test_t_sne.py::test_auto_learning_rate[500]**
5.31s call     ensemble/tests/test_common.py::test_ensemble_heterogeneous_estimators_behavior[stacking-classifier]
5.25s call     ensemble/tests/test_bagging.py::test_classification
...

Reply:
I see. I don't know how to make it faster so I removed the test.